index.js

```diff
@@ -70,6 +73,30 @@ class Conf {
 	clear() {
 		this.store = obj();
 	}
+	onDidChange(key, callback) {
+		if (typeof key !== 'string') {
+			throw new TypeError(`Expected \`key\` to be of type \`string\`, got ${typeof key}`);
+		} else if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		let currentValue = key ? this.get(key) : this.store;
+
+		const onChange = () => {
+			const oldValue = currentValue;
+			const newValue = key ? this.get(key) : this.store;
+
+			try {
+				assert.deepEqual(newValue, oldValue);
+			} catch (err) {
+				currentValue = newValue;
+				callback.call(this, newValue, oldValue, key);
```

Comment:
It doesn't make sense to have `key` as the third argument when the user already knows about it, IMO.